### PR TITLE
WT-6793 Moved some details from Diary's narrative to notes

### PR DESCRIPTION
**Pull request by alvinllobrera**

### CODE REVIEW

**Comment by vyshane** [comment]
Can you add a test to assert that any reasonable notes content that we produce will be within the maximum number of characters supported?

File: src/main/java/io/wisetime/connector/inprotech/InprotechDao.java [hunk 1]
```diff
@@ -166,7 +166,8 @@
    *                                    between @pdtStartTime and @pdtEndTime
    *   @pdtStartTime datetime = null, -- in yyyy-MM-dd HH:mm:ss:SSS format
    *   @pdtEndTime datetime = null, -- in yyyy-MM-dd HH:mm:ss:SSS format
-   *   @psNarrative nvarchar(max) = null
+   *   @psNarrative nvarchar(max) = null,
+   *   @psNotes nvarchar(254)   = null
```
Comment (vyshane) [inline src/main/java/io/wisetime/connector/inprotech/InprotechDao.java]: What happens if the stored procedure receives more than 254 characters? Ideally we want an error to be logged by the connector. Should we add a precondition check at the start of this method?

File: src/main/java/io/wisetime/connector/inprotech/InprotechConnector.java [hunk 2]
```diff
@@ -167,13 +174,19 @@ public PostResult postTime(Request request, TimeGroup userPostedTimeUtc) {
     // The important information is the time spent, which we set as the (possibly edited) total time.
     final LocalDateTime endTime = activityStartTime.get().plus(workedTime, ChronoUnit.SECONDS);
 
+    final String narrativeMsg = narrativeTemplateFormatter.format(zonedTimeGroup);
+    final String notesMsg = internalNotesTemplateFormatter.format(zonedTimeGroup);
+    if (notesMsg.length() > NOTES_MAX_LENGTH) {
+      return PostResult.PERMANENT_FAILURE.withMessage("Maximum length for notes exceeded.");
```
Comment (alvinllobrera) [inline src/main/java/io/wisetime/connector/inprotech/InprotechConnector.java]: Fail early if generated notes exceeded max length

File: src/main/java/io/wisetime/connector/inprotech/InprotechDao.java (hunk 1 above)
Comment (alvinllobrera) [reply src/main/java/io/wisetime/connector/inprotech/InprotechDao.java]: The stored procedure will fail. I have updated the PR to check if notes exceeded its max length

File: src/main/resources/inprotech-internal-notes-template.ftl [hunk 3]
```diff
@@ -0,0 +1,14 @@
+${'\r\n'}Total Worked Time: ${getTotalDuration(getTimeRows())?string.@duration}
+Total Chargeable Time: ${getTotalDurationSecs()?string.@duration}
+Experience Weighting: ${getUser().getExperienceWeightingPercent()}%
+<#if getDurationSplitStrategy() == "DIVIDE_BETWEEN_TAGS" && (getTags()?size > 1)>
```
Comment (vyshane) [inline src/main/resources/inprotech-internal-notes-template.ftl]: This text refers to "above times", which aren't in the notes, but are in the narrative. Let's move this section to the narrative.

Comment (alvinllobrera) [reply src/main/resources/inprotech-internal-notes-template.ftl]: I thought this was referring to the **Total Worked and Chargeable Time** which do not match to the duration on time row info on narratives because the `narrative` always displays the whole duration of each time row while the duration on the `notes` is divided.

Comment (vyshane) [reply src/main/resources/inprotech-internal-notes-template.ftl]: Correct, but the rows are not displayed in the notes. They are in the narratives.

Comment (alvinllobrera) [reply src/main/resources/inprotech-internal-notes-template.ftl]: Ok, PR updated

**Comment by vyshane** [comment]
LGTM
